### PR TITLE
Fix disabling auto-update for a single DataSource

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
User can remove the label 'cdi.kubevirt.io/dataImportCron' from a DataSource to disable auto-updates.

**Which issue(s) this PR fixes**: 
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2051693

**Release note**:
```release-note
None
```

### CODE REVIEW

File: tests/dataSources_test.go [hunk 1]
```diff
@@ -936,27 +1000,86 @@ var _ = Describe("DataSources", func() {
...
+						return apiClient.Update(ctx, ds)
+					}, shortTimeout, time.Second).Should(Succeed())
+
+					Eventually(func() metav1.StatusReason {
+						err := apiClient.Get(ctx, dataImportCron.GetKey(), dataImportCron.NewResource())
+						return errors.ReasonForError(err)
+					}, 2*shortTimeout, time.Second).Should(Equal(metav1.StatusReasonNotFound), "DataImportCron should not exist.")
+				})
+
+				It("[test_id:TODO] should restore DataSource, when CDI label is removed", func() {
```
Comment (ksimon1) [inline tests/dataSources_test.go:1053]: it looks like this test might be flaky

**Comment by akrejcir** [comment]
The CI is flaky, because it uses latest released CDI version, which does no yet include PR https://github.com/kubevirt/containerized-data-importer/pull/2142 .

With the PR, the tests should be stable.

**Comment by akrejcir** [comment]
I will verify it on my cluster with the CDI PR.
/hold

**Comment by akrejcir** [comment]
The tests pass in my env.

/override ci/prow/e2e-functests
/override ci/prow/e2e-upgrade-functests

**Pull request merged**